### PR TITLE
Broadcasting category changes only to the correct user

### LINKED ISSUE #2707: Permissions Branch: New sidebar categories get added for everyone
## Steps to reproduce the behavior

1. Go to any category in the new sidebar
2. Select the  "..." menu
3. Click on "Create New Category"
4. Name the category
5. Log into to a different account
6. See the new category in users sidebar
7. Switching to Channels then back to Boards fixes this and only displays the categories the user should be seeing.

## Expected behavior

Categories should only be visible to the user who created them, and should not get added to another user's sidebar.

## Screenshots (optional)
| User A  | User B |
|--|--|
| Screen Shot 2022-04-01 at 12 29 37 PM | Screen Shot 2022-04-01 at 12 29 24 PM |

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: permissions-feature-branch
 - Browser and OS: Chrome on Mac

### PR DESCRIPTION
**Pull request by harshilsharma63**
Fixes #2707

### CODE REVIEW

File: server/ws/plugin_adapter.go [hunk 1]
```diff
@@ -476,7 +480,7 @@ func (pa *PluginAdapter) BroadcastCategoryChange(category model.Category) {
 		Category: &category,
 	}
 
-	pa.sendTeamMessage(websocketActionUpdateCategory, category.TeamID, utils.StructToMap(message))
+	pa.sendUserMessageSkipCluster(websocketActionUpdateCategory, utils.StructToMap(message), category.UserID)
```
Comment (jespino) [inline server/ws/plugin_adapter.go:483]: Why are we skipping the cluster here? I don't think we should skip the cluster propagation here.

**Review by sbishel — APPROVED** [review]
@harshilsharma63 - Other than Jesus comment, LGTM.

File: server/ws/plugin_adapter.go (hunk 1 above)
Comment (harshilsharma63) [reply server/ws/plugin_adapter.go]: Added now.

**Comment by harshilsharma63** [comment]
@sbishel can you re-review my last change? Add good to go then.

**Pull request merged**